### PR TITLE
add notes for n25768

### PR DESCRIPTION
**Pull request by glozow**

### CODE REVIEW

File: _posts/2022-08-24-#25768.md [hunk 1]
```diff
@@ -7,18 +7,101 @@ authors: [achow101]
...
+### Approach
+
+1. Can you think of any other methods of getting the wallet transactions sorted in this order?
+
+1. Would it make sense to *keep* the wallet transactions sorted in this order? Why is `mapWallet`
+   implemented as a `std::unordered_map`; would another data structure make sense?
+
+1. If you thought of an alternative approach, how does it compare? What is its runtime complexity?
+   How much {more, less} memory is allocated? Does it require a lot of code to be
+(re)written? Is its complexity appropriate for its purpose?
```
Comment (stickies-v) [inline _posts/2022-08-24-#25768.md:76]:
```suggestion
(re)written? Is its complexity appropriate for this purpose?
```

File: _posts/2022-08-24-#25768.md [hunk 2]
```diff
@@ -7,18 +7,101 @@ authors: [achow101]
...
+
+### Implementation
+
+1. How is `GetSortedTxs()`, the function extracted from `ReacceptWalletTransactions()`, implemented?
+   How much memory is needed, as a function of the transactions in `mapWallet`?
+
+1. `GetSortedTxs()` is
+   [declared](https://github.com/bitcoin-core-review-club/bitcoin/commit/9a556564e9dc64ae0ad723c78da33d0c982f006f#diff-9ce137cd784ea308778842120aa2af6d2bb8369485b71f25e72b2a32cf0a5b21R536)
+as requiring the wallet lock to already be held. Why or why not does this make sense?
```
Comment (stickies-v) [inline _posts/2022-08-24-#25768.md:85]:
```suggestion
as requiring the wallet lock to already be held. Why does this (not) make sense?
```

File: _posts/2022-08-24-#25768.md [hunk 3]
```diff
@@ -7,18 +7,101 @@ authors: [achow101]
...
+
+1. A [structured binding declaration](https://en.cppreference.com/w/cpp/language/structured_binding) is used in
+   [two](https://github.com/bitcoin-core-review-club/bitcoin/commit/9a556564e9dc64ae0ad723c78da33d0c982f006f#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R1826)
+[places](https://github.com/bitcoin-core-review-club/bitcoin/commit/9a556564e9dc64ae0ad723c78da33d0c982f006f#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R1831)
+in `GetSortedTxs()`. What are the types of each variable (`wtxid`, `wtx`, `_`, and `wtx`) and what
+do they represent? Why or why not is this better than declaring the types explicitly?
```
Comment (stickies-v) [inline _posts/2022-08-24-#25768.md:91]:
```suggestion
do they represent? Why is this (not) better than declaring the types explicitly?
```

File: _posts/2022-08-24-#25768.md [hunk 4]
```diff
@@ -7,18 +7,101 @@ authors: [achow101]
...
+
+1. Can you find any behavior changes to `ReacceptWalletTransactions()`, or is it a move-only code
+   change?
+
+1. The test
+   [calls](https://github.com/bitcoin-core-review-club/bitcoin/commit/e40ddf36bed81bdf28d386eb961c9ed22b69e207#diff-2dd85d481900d4ad19d113d2114861b0134bcd283435e95b18d10adf5ad381a0R112-R113)
+both `setmocktime` and `mockscheduler` to trigger a rebroadcast. What is the difference between
+these two calls, and why or why not is it necessary to call both of them?
```
Comment (stickies-v) [inline _posts/2022-08-24-#25768.md:99]:
```suggestion
these two calls, and why is it (not) necessary to call both of them?
```

**Review by stickies-v — APPROVED** [review]
ACK c538c79d69380c93c441418fbcf550a7fb7a9939 - very comprehensive and easy to follow!

Left some minor suggestions but I don't feel strongly about them at all.

File: _posts/2022-08-24-#25768.md (hunk 1 above)
Comment (glozow) [reply _posts/2022-08-24-#25768.md]: Done

File: _posts/2022-08-24-#25768.md (hunk 2 above)
Comment (glozow) [reply _posts/2022-08-24-#25768.md:85]: I prefer "why or why not" tbh

Comment (stickies-v) [reply _posts/2022-08-24-#25768.md:85]: Okay then keep it. I feel like "Why or why not?" works well in a separate sentence but "why not does this" sounds a bit weird to me. But no strong feelings about it so whatever you prefer - not worth spending too much time on!

Comment (glozow) [reply _posts/2022-08-24-#25768.md:85]: Yeah makes sense, thanks for the feedback! Mostly did it habitually, hopefully it's not too confusing for people heh.

**Pull request merged**